compass/landice/tests/mismipplus/setup_mesh.py

```diff
@@ -0,0 +1,301 @@
+import warnings
+
+import numpy as np
+import xarray as xr
+from mpas_tools.io import write_netcdf
+from mpas_tools.logging import check_call
+from mpas_tools.mesh.conversion import convert, cull
+from mpas_tools.planar_hex import make_planar_hex_mesh
+
+from compass.model import make_graph_file
+from compass.step import Step
+
+
+class SetupMesh(Step):
+    """
+    A step for creating a mesh and initial condition for MISMIP+ test cases
+
+    Attributes
+    ----------
+    resolution : int
+        The resolution of the test case, as defined configuration file at the
+        when `compass setup` is run.
+
+    nx : int
+        The number of cells in the x direction
+
+    ny : int
+        The number of cells in the y direction
+
+    dc : int
+        The distance in meters between adjacent cell centers.
+
+    """
+    def __init__(self, test_case, resolution):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        testcase : compass.TestCase
+            The test case this step belongs to
+
+        resolution : int
+            The distance between horizontal grid points
+        """
+        # Defined resolutions, the value of `resolution` in the configuration
+        # file needs to a key in the dictionary, or added manually.
+        resolution_params = {'8km': {'nx': 73, 'ny': 14, 'dc': 9237.604307},
```

Comment:
@trhille , for your usage of this domain for convergence testing with calving on, you will presumably want a gutter beyond the prescribed initial ice shelf calving front to allow for portions of the calving front to advance under prognostic calving without hitting the edge of the mesh.  My recollection is the `nx` values specified here give 2 extra cells (at all resolutions) beyond the MISMIP+ fixed calving front position.   

If allowing a gutter option sounds desirable now, I think that would be a new cfg option, either with units of number of cells or distance.  Do you want that included in this PR so you can use it out of the gate for your planned tests?

Reply:
Yes, good idea. I don't have a strong preference for whether it should be specified in number of cells or distance, but it should end up being ~the same distance for all resolutions, so maybe specifying distance is the simplest options.